include/eosio/vm/backend.hpp

```diff
@@ -130,6 +131,16 @@ namespace eosio { namespace vm {
          construct();
       }
 
+      backend& operator=(const backend& other) {
+         if (this != &other) {
+            mod                    = other.mod;
```

Comment:
Where is this used. I'm not understanding the use of `shared_ptr` over `unique_ptr` and why we need a copy that shares `module`.

Reply:
It is used https://github.com/AntelopeIO/leap/blob/47e0372ab257534ae5e4133e3e3c6f894569125e/libraries/chain/webassembly/runtimes/eos-vm.cpp#L133. `mod` stores compiled information about a WASM; it is shared across multiple read-only threads when those threads happen to execute the same contract. The other fields `exec_ctx_created_by_backend`, `initial_max_call_depth`, and `initial_max_pages` are initial values of a backend (instantiated module). They are needed at the start of each execution.

Reply:
Personally I would prefer a `clone()` method, or maybe in this case `share()`. For such a complicated type using `operator=` in my opinion is too error prone.